Rewrite to use an external data file, and generally reorganize into a more regimented data flow.

tabatkins opened the pull request:
(Fixes #2.)

The new data format is a line-based format, which should be easy to read and PR.

Each line looks like `+org: w3c`:

* you can use `+` to add things to the set of explored stuff, or `-` to skip them
* you can specify `org`, `repo`, or `file`, with the values after being the path segment following after `github.com`

(Currently `-org` isn't supported, since it's useless, and `+file` isn't supported because I haven't figured it out yet.)

Whitespace around the line is ignored; I'm using indentation as a readability aid, to nest skipped repos/files under the org we've added.

Comments are also allowed - any line starting with a `#` is a comment and is ignored.  Blank lines are also ignored.

(This should probably be written into a README.)

---

The overall data flow has also been more regimented - first I walk all the orgs to find repos we should search, then we search all the repos to find files we should grab, then we grab all the files.  This is done in three distinct phases to make it easier to add arbitrary repos/files, which seems like it should probably be useful.

tabatkins committed:
Rewrite to use an external data file, and generally reorganize into a more regimented data flow.

tabatkins commented:
I'm not sure what's going on in the Travis build; the error appears to be something unrelated to actually running the code.

foolip commented on update.py:
```diff
@@ -1,57 +1,118 @@
         raise
-    skip_files = SKIP_FILES.get(repo.full_name)
     for entry in tree.tree:
+        if entry.path in skipFiles:
```
I tried running before/after your changes and the only difference is that w3c/fxtf-drafts/filter-effects/Overview.bs came back because of a bug on this line. Will push a fix.

foolip committed:
Combine repo and file paths before comparing to skipFiles

The alternative to this would be to make specs.data indentation be
meaningful, but that seems like overkill.

foolip committed:
Add newline to specs.data

foolip commented:
Thanks! I've pushed a fix that makes the output identical.

I've also disabled Travis on PRs since it doesn't do anything useful, the config is just for master.
